datafs/core/data_api.py

```diff
@@ -169,6 +169,67 @@ def get_archive(self, archive_name, default_version=None):
             default_version=default_version,
             **res)
 
+    def listdir(self, location, authority_name=None):
+        '''
+        List archive path components at a given location
+
+        .. Note ::
+
+            When using listdir on versioned archives, listdir will provide the
+            version numbers when a full archive path is supplied as the
+            location argument. This is because DataFS stores the archive path
+            as a directory and the versions as the actual files when versioning
+            is on.
+
+        Parameters
+        ----------
+
+        location : str
+
+            Path of the "directory" to search
```

Comment:
Sort of, although this reveals the aspect I'm a little squeamish about. `listdir` here is simply exposing the [listdir method](http://docs.pyfilesystem.org/en/latest/reference/base.html#fs.base.FS.listdir) of the underlying Pyfilesystem `fs` object. So the path is interpreted by `fs`, not by `DataFS`. No convention about it. In our case, this is actually listing the virtual directory on OSDC's s3 storage.

I agree, though, that an article in the docs regarding archive naming conventions would be helpful.